container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/backend_creator/backend_creator.go

```diff
@@ -73,10 +78,35 @@ func getLocalDockerKurtosisBackend(
 	optionalApiContainerModeArgs *APIContainerModeArgs,
 ) (backend_interface.KurtosisBackend, error) {
 	dockerClientOpts := []client.Opt{
-		client.FromEnv,
 		client.WithAPIVersionNegotiation(),
 	}
 
+	// If the DOCKER_HOST env variable is set, use it. Otherwise, try to locate the daemon socket. Otherwise, fall back
+	// to env variables as we were doing before
+	userHomeDir, err := os.UserHomeDir()
+	if err != nil {
+		logrus.Debugf("Unable to locate user's home directory, this might affect connection to docker.")
+	}
+	userOwnedSocketPath := fmt.Sprintf("%s%s", userHomeDir, userOwnDaemonSocket)
+	if dockerHostEnvVar := os.Getenv(client.EnvOverrideHost); dockerHostEnvVar != "" {
+		logrus.Debugf("Connecting to Docker daemon at '%s'", dockerHostEnvVar)
+		dockerClientOpts = append(dockerClientOpts, client.WithHostFromEnv())
+	} else if _, err := os.Stat(systemDaemonSocket); err == nil {
+		logrus.Debugf("Connecting to Docker daemon via unix socket '%s'", systemDaemonSocket)
+		fullyQualifiedUnixSocket := fmt.Sprintf("%s%s", unixSocketPrefix, systemDaemonSocket)
+		dockerClientOpts = append(dockerClientOpts, client.WithHost(fullyQualifiedUnixSocket))
+	} else if _, err := os.Stat(userOwnedSocketPath); err == nil {
+		logrus.Debugf("Connecting to Docker daemon via unix socket '%s'", userOwnedSocketPath)
+		fullyQualifiedUnixSocket := fmt.Sprintf("%s%s", unixSocketPrefix, userOwnedSocketPath)
+		dockerClientOpts = append(dockerClientOpts, client.WithHost(fullyQualifiedUnixSocket))
+	} else {
+		logrus.Debugf("Unable to locate Docker daemon socket and '%s' environment variable wasn't set. Falling "+
+			"back to Docker's own way to connect to locally running daemon. If it fails, make sure docker is running "+
+			"locally and try setting '%s' to help Kurtosis connect to it.", client.EnvOverrideHost,
+			client.EnvOverrideHost)
+		dockerClientOpts = append(dockerClientOpts, client.FromEnv)
```

Comment:
Trusting this has been discussed somewhere, but the previous default way is now the final fallback, so people who might have the above configured will get some different behavior than before maybe?

I suspect this is an extreme edge case and isn't much to worry about

Reply:
I think it's the "new" expected behaviour yes, at least reading for the the associated tickets. Basically we is DOCKER_HOST is set, we want to use it in priority. Which I find also more comprehensive to users. If they made the effort of setting this env variable, it's probably for something and we should default to using it. I won't the here to roll back if this change causes issues, but I guess it's fairly small and can be reverted easily :)